lib/msal-core/test/UserAgentApplication.spec.ts

```diff
@@ -1989,16 +1987,19 @@ describe("UserAgentApplication.ts Class", function () {
 
         it("tests user query string present on final page url and token response is cached", function() {
             config.auth.navigateToLoginRequestUrl = true;
-            const loginUrl = "http://localhost:8081/test/"
-            const userQueryString = "?testKey=testVal"
+            const loginUrl = "http://localhost:8081/test/";
+            const userQueryString = "?testKey=testVal";
             const loginStartPage = loginUrl + userQueryString;
             const successHash = testHashesForState(TEST_LIBRARY_STATE).TEST_SUCCESS_ID_TOKEN_HASH + TEST_USER_STATE_NUM;
 
             window.location.href = loginStartPage;
             window.location.search = userQueryString;
 
             sinon.stub(window, "parent").returns(window);
-            sinon.stub(window.location, "href").returns(loginStartPage + successHash)
+            sinon.stub(window.location, "href").returns(loginStartPage + successHash);
+            sinon.stub(window.history, "replaceState").callsFake((data, title, url) => {
+                window.location.hash = "";
```

Comment:
Not really sure of a better way to handle jsdom not fully supporting the `history` APIs.